tests_and_analysis/test/euphonic_test/test_spectrum1dcollection.py

```diff
@@ -419,3 +432,158 @@ def test_broaden(self, spectrum_file, width, shape):
+             {'top_level_key': 'something', 'top_level_int': 10,
+              'line_data': [
+                  {'sample': 0, 'inst': 'LET', 'index': 10, 'other_data': 'misc'},
+                  {'sample': 2, 'inst': 'MARI', 'some_other_data': 5},
+                  {'sample': 0, 'inst': 'MAPS', 'index': 10, 'other_data': 'another_value'},
+                  {'sample': 0, 'inst': 'TOSCA'},
+                  {'sample': 2, 'inst': 'LET'},
+                  {'sample': 1, 'inst': 'LET', 'index': 10}]}),
+            ('quartz_666_pdos.json', fake_metadata,
+             'index',
+             {'top_level_key': 'something', 'top_level_int': 10,
+              'line_data': [
+                  {'index': 10},
+                  {'sample': 2, 'index': 5},
+                  {'sample': 0, 'inst': 'TOSCA', 'index': 7}]})
+            ])
+    def test_group_by_fake_metadata(
+            self, spectrum_file, metadata, group_by_args, expected_metadata):
+        spec_col = get_spectrum1dcollection(spectrum_file)
+        spec_col.metadata = metadata
+        grouped_spec = spec_col.group_by(group_by_args)
+        assert  grouped_spec.metadata == expected_metadata
+
+    # Self-consistency test, allows us to test more metadata combinations
+    # without generating more test files
+    @pytest.mark.parametrize(
+        'spectrum_file, metadata, group_by_args, group_indices', [
+            ('quartz_666_pdos.json', fake_metadata, ['sample', 'inst'],
+             [[0], [1, 2], [3], [4,5], [6,7], [8]]),
+            ('quartz_666_pdos.json', fake_metadata, 'index',
+             [[0, 2, 3, 5, 6, 8], [1, 7], [4]])])
+    def test_group_by_same_as_index_and_sum_with_fake_metadata(
+            self, spectrum_file, metadata, group_by_args, group_indices):
+        spec_col = get_spectrum1dcollection(spectrum_file)
+        spec_col.metadata = metadata
+        grouped_spec = spec_col.group_by(group_by_args)
+        for i, spec in enumerate(grouped_spec):
+            check_spectrum1d(spec,
+                             spec_col[group_indices[i]].sum())
+
+    @pytest.mark.parametrize(
+        'spectrum_file, select_kwargs, selected_spectrum_file', [
+            ('methane_pdos.json', {'label': ['H2', 'H3', 'H4']},
+             'methane_pdos_index_1_4.json'),
+            ])
+    def test_select(self, spectrum_file, select_kwargs,
+                    selected_spectrum_file):
+        spec_col = get_spectrum1dcollection(spectrum_file)
+        selected_spec = spec_col.select(**select_kwargs)
+        expected_selected_spec = get_expected_spectrum1dcollection(
+            selected_spectrum_file)
+        check_spectrum1dcollection(selected_spec, expected_selected_spec)
+
+    # Keys that were initially in the high-level metadata dict (e.g.
+    # weighting from calculate_pdos) are put into 'line_data' when
+    # two Spectrum1DCollection objects are added together, and are
+    # not put back into the high-level dict again by select, which
```

Comment:
I guess I just wasn't sure if we wanted to change the behaviour of `getitem` since it had already been implemented, other than that no reason really! I'll have a go at putting some of this logic in

Reply:
It was actually much easier than anticipated... done!

Reply:
Only because you'd already done 90% of the work! Huzzah for structured programming.